staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -453,11 +455,13 @@ func (c *Config) Complete(informers informers.SharedInformerFactory) CompletedCo
 	// if there is no port, and we listen on one securely, use that one
 	if _, _, err := net.SplitHostPort(c.ExternalAddress); err != nil {
 		if c.SecureServing == nil {
-			klog.Fatalf("cannot derive external address port without listening on a secure port.")
+			klog.ErrorS(nil, "Cannot derive external address port without listening on a secure port")
+			os.Exit(1)
```

Comment:
from https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

Fatal calls use a default exit code of 255. When migrating, please use an exit code of 1 and include an "ACTION REQUIRED:" release note.

Reply:
Thank you very much for your comments, updated.